### PR TITLE
fix multi-root duplicate relative paths

### PR DESCRIPTION
**Pull request by Samsen879**
## Summary

  This fixes a multi-root edge case where files with the same relative path could be handled incorrectly.

  For example, if two roots both contain `README.md` or `src/index.ts`, the current packing flow can duplicate those files in XML/
  Markdown/plain output. In JSON output, the later file can overwrite the earlier one because both entries use the same key.

  The fix keeps each root's file list separate during collection, then disambiguates file paths for multi-root duplicate relative
  paths.

  ## Changes

  - Deduplicate and sort file paths per root instead of flattening everything first.
  - Use root-aware display paths for multi-root output, like `packages/a/README.md` and `packages/b/README.md`.
  - Keep the existing JSON shape as `files: Record<string, string>`.
  - Add regression tests for duplicate relative paths across multiple roots.

  ## Checklist

  - [x] Run `npm run test`
  - [x] Run `npm run lint`

  ## Test Plan

  - `npm test tests/core/packager/multiRootSpec.test.ts`
  - `npm test`
  - `npm run build`
  - `npm run lint` exits successfully; it reports three pre-existing warnings outside this PR's scope.

### CODE REVIEW

**Comment by Samsen879** [comment]
Updated the branch to address the review comments:

- roots outside cwd now use basename + suffixes instead of walking absolute parent directories, avoiding host/local parent directory names in output paths
- the fallback now suffixes the last, most-disambiguated candidate labels
- the JSON regression now checks the outside-cwd same-basename case with app/app-2 keys

**Comment by Samsen879** [comment]
@coderabbitai review

**Comment by yamadashy** [comment]
Thanks for the PR! I'll take a look.

File: src/core/packager.ts [hunk 1]
```diff
@@ -65,6 +65,87 @@ export interface PackOptions {
...
+      const segments = rootDirSegments[index];
+      return segments.slice(-depth).join('/') || label || `root-${index + 1}`;
+    });
+
+    if (new Set(candidates).size === candidates.length) {
+      return candidates;
+    }
+  }
+
+  return uniquifyLabelsWithSuffixes(labels);
```
[the remark this answers is not in the document]
Comment (yamadashy) [reply src/core/packager.ts]: No longer applicable — the depth-based disambiguation loop this referred to was removed. `buildRootLabels` now reduces to a label (cwd-relative path, or basename for roots outside cwd) plus a numeric suffix on collision, so there are no longer partial candidates to preserve. 🤖

**Review by yamadashy — APPROVED** [review]
